### ISSUE TITLE
Error when querying Hive partitioned table on Google Cloud Storage (GCS)

### ISSUE THREAD

**Issue by gddezero** [StarRocks/starrocks#18912]
Querying Hive **non-partitioned** table on GCS works fine. But querying Hive **partitioned** table on Google Cloud Storage (GCS) will fail.

### Steps to reproduce the behavior (Required)
1. Install GCS connector to FE/lib
wget https://storage.googleapis.com/hadoop-lib/gcs/gcs-connector-hadoop3-latest.jar -P $STARROCKS_HOME/fe/lib
2. Create a Hive catalog
CREATE EXTERNAL CATALOG hive_catalog_hms
PROPERTIES
(
    "type" = "hive",
    "aws.s3.use_instance_profile" = "false",
    "aws.s3.enable_ssl" = "false",
    "aws.s3.enable_path_style_access" = "true",
    "aws.s3.access_key" = "< GCS Access Key >",
    "aws.s3.secret_key" = "< GCS Secret Key >",
    "aws.s3.endpoint" = "storage.googleapis.com",
    "aws.s3.region" = "us-central1",
    "hive.metastore.uris" = "thrift://< IP or hostname of Hive Metastore >:9083"
);
3. Query hive  partitioned table
use hive_catalog_hms.< database >;
select * from < partitioned table > limit 10;

### Expected behavior (Required)
The query should success without error

### Real behavior (Required)
The query failed with error message:
```
ERROR 1064 (HY000): code=403(SdkErrorType:22), message=The request signature we calculated does not match the signature you provided. Check your Google secret key and signing method.


I0221 07:36:34.147534 3307570 fragment_executor.cpp:158] Prepare(): query_id=774027c1-b1ba-11ed-a93b-024213766ee0 fragment_instance_id=774027c1-b1ba-11ed-a93b-024213766ee2 backend_num=0
I0221 07:36:34.149526 3307564 fragment_executor.cpp:158] Prepare(): query_id=774027c1-b1ba-11ed-a93b-024213766ee0 fragment_instance_id=774027c1-b1ba-11ed-a93b-024213766ee1 backend_num=1
I0221 07:36:34.150056 3307472 hdfs_scanner.cpp:151] open file success: gs://forrest-tpch/sf1000/parquet/test_partitioned/dt=2023-01-02/000000_0
I0221 07:36:34.150540 3307444 hdfs_scanner.cpp:151] open file success: gs://forrest-tpch/sf1000/parquet/test_partitioned/dt=2023-01-01/000000_0
W0221 07:36:34.175235 3307472 hdfs_scanner_text.cpp:243] Status is not ok code=403(SdkErrorType:22), message=The request signature we calculated does not match the signature you provided. Check your Google secret key and signing method.
E0221 07:36:34.175277 3307472 hdfs_scanner.cpp:134] failed to read file: gs://forrest-tpch/sf1000/parquet/test_partitioned/dt=2023-01-02/000000_0

W0221 07:36:34.175367 3307389 pipeline_driver.cpp:226] pull_chunk returns not ok status IO error: code=403(SdkErrorType:22), message=The request signature we calculated does not match the signature you provided. Check your Google secret key and signing method.
/root/starrocks/be/src/exec/vectorized/hdfs_scanner.cpp:23 value_or_err_L23
/root/starrocks/be/src/exec/vectorized/hdfs_scanner_text.cpp:92 value_or_err_L92
/root/starrocks/be/src/formats/csv/csv_reader.cpp:19 _fill_buffer()
/root/starrocks/be/src/exec/vectorized/hdfs_scanner_text.cpp:67 CSVReader::next_record(record)
/root/starrocks/be/src/exec/vectorized/hdfs_scanner_text.cpp:194 parse_csv(runtime_state->chunk_size(), chunk)
/root/starrocks/be/src/connector/hive_connector.cpp:405 _scanner->get_next(state, chunk)
/root/starrocks/be/src/exec/pipeline/scan/scan_operator.cpp:197 _get_scan_status()
W0221 07:36:34.175385 3307389 pipeline_driver_executor.cpp:139] [Driver] Process error, query_id=774027c1-b1ba-11ed-a93b-024213766ee0, inst
ance_id=774027c1-b1ba-11ed-a93b-024213766ee1, status=IO error: code=403(SdkErrorType:22), message=The request signature we calculated does 
not match the signature you provided. Check your Google secret key and signing method.
/root/starrocks/be/src/exec/vectorized/hdfs_scanner.cpp:23 value_or_err_L23
/root/starrocks/be/src/exec/vectorized/hdfs_scanner_text.cpp:92 value_or_err_L92
/root/starrocks/be/src/formats/csv/csv_reader.cpp:19 _fill_buffer()
/root/starrocks/be/src/exec/vectorized/hdfs_scanner_text.cpp:67 CSVReader::next_record(record)
/root/starrocks/be/src/exec/vectorized/hdfs_scanner_text.cpp:194 parse_csv(runtime_state->chunk_size(), chunk)
/root/starrocks/be/src/connector/hive_connector.cpp:405 _scanner->get_next(state, chunk)
/root/starrocks/be/src/exec/pipeline/scan/scan_operator.cpp:197 _get_scan_status()
```

### StarRocks version (Required)
+-------------------+
| current_version() |
+-------------------+
| 2.5.2 c3772fb     |
+-------------------+

### Root Cause
The object name of the partition table contain sepcial character of '='. StarRocks depens on aws-sdk-cpp to request object on GCS. Currently StarRocks is build with aws-sdk-cpp 1.9.179 which does not encode special characters properly for GCS. Check the details in Github issue: [Objects with "=" in name cause 403 error on some providers. #1224](https://github.com/aws/aws-sdk-cpp/issues/1224)

### Solution
Upgrade aws-sdk-cpp to at least 1.9.272

**Comment by jaogoy** [comment]
Does StarRocks support GCS already?

**Comment by gddezero** [comment]
> Does StarRocks support GCS already?

Yes. But through S3 compatible API to read from GCS. I wrote an article for how to query Hive table on GCS using StarRocks: https://github.com/gddezero/gcp-samples/tree/main/starrocks_gcs

**Comment by kevincai** [comment]
aws sdk 1.9.272 contains a serious bug https://github.com/aws/aws-sdk-cpp/issues/1973 which is fixed in [1.9.320](https://github.com/aws/aws-sdk-cpp/releases/tag/1.9.320)

Can we postpone this fix until find the next stable aws sdk version?

**Comment by gddezero** [comment]
> aws sdk 1.9.272 contains a serious bug [aws/aws-sdk-cpp#1973](https://github.com/aws/aws-sdk-cpp/issues/1973) which is fixed in [1.9.320](https://github.com/aws/aws-sdk-cpp/releases/tag/1.9.320)
> 
> Can we postpone this fix until find the next stable aws sdk version?

Thanks for pointing out the aws sdk bug. I also heard that native GCS support is also under development, which means we don't need to upgrade aws-sdk-cpp until next stable aws sdk version.

**Comment by Smith-Cruise** [comment]
This problem has been solved, you can add `aws_sdk_enable_compliant_rfc3986_encoding=true` in be.conf to solve this problem.
